### ISSUE TITLE
Openjdk-style: new check UpperCaseLetterInHexadecimalIntLiteralCheck

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#16836]
![Image](https://github.com/user-attachments/assets/b080ef3b-0568-4532-b66f-ef13d473efcf)
(Second Rule)

very similar to [UpperEllCheck](https://github.com/checkstyle/checkstyle/blob/17faf4b60c4c3237097f9c22cac8ca356db0baeb/src/main/java/com/puppycrawl/tools/checkstyle/checks/UpperEllCheck.java#L55), `UpperCaseLetterInHexadecimalIntLiteralCheck` should check that the 
hexademical num literals (any literal which starts with `0x`) contain lowercase letters for hexadecimal letters (A, B, C, D, E, F)
Check will violate on all the following tokens if they use lower-case hexa-letters 
- NUM_DOUBLE
- NUM_FLOAT
- NUM_LONG
- NUM_INT

```java
public class Literals {

    public static void main(final String[] args) {

        byte b1  = 0x1A;                     // ok
        byte b2  = 0x1b;                     // violation, 0x1B;

        short s1 = 0xA8C;                   // ok
        short s2 = 0xF5f;                   // violation, 0xF5F

        int i1 = 0x11 + 0xabc;                // violation, 0xABC
        int i2 = 0X123 + 0xabc;               // violation, 0xABC
        int i3 = 0xdeadbeef;                  // violation, 0xDEADBEEF

        long l1 = 0x7fff_ffff_ffff_ffffL;     // violation, lowercased f's
        long l2 = 0x7FFF_AAA_bBB_DDDL;        // violation, lowercase b

        float f2 = 0x1.2p3f;                  // no violation, 
        float f3 = 0x3.4fp0F;                 // violation, 0x3.4Fp0F
        float f4 = 0.123E4f;                  // no violation

        double d1 = 0xA.0p2;                  // no violation 
        double d2 = 0xA1B2.5p1;               // no violation
        double d3 = 0x1.2dp2D;                // violation, 0x1.2Dp2D;
        double d4 = 0x1.2ADp2d;               // no violation
    }
}
```

it should produce the following output when run on above file

```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="UpperCaseLetterInHexadecimalIntLiteralCheck"/> 
 </module> 
</module>
```

```
Starting audit...
[WARN] /mnt/5D92528E6B945467/test/Literals.java:5: Hexademical Literal contains lowercase letters.[UpperCaseLetterInHexadecimalIntLiteralCheck]
Audit done.
```

**Comment by romani** [comment]
Why description is focused on uppercase , while style guide tells "do lowercase for all except long type 'l'" ?

name is challange, i do do no know how to name it in good way.  "HexadecimalInt" is not suites for float/double.

https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-3.10.2
>3.10.2. Floating-Point Literals
A floating-point literal has the following parts: a whole-number part, a decimal or hexadecimal point (represented by an ASCII period character), a fraction part, an exponent, and a type suffix.

**Comment by mohitsatr** [comment]
@romani style guide tells "do lower case for all except Long(L) suffix **and** hexademical letters(A-F)"

how about `UppercaseHexLiteralCheck` ?

**Comment by romani** [comment]
ok, we have two options:
to name it with target of what we catch (put violation on) in code
to name it as what is expected as good

https://checkstyle.sourceforge.io/checks.html

I see pattern "Avoid...." , "Illegal..."

What if we name it `CaseForHexLiteralCheck` without defining lower or upper, and may be in far future it will be property to let user demand upper case if they need. For now we will demand all to be lowercase, except for `l`.

**Comment by mohitsatr** [comment]
@romani let me know if there's anything missing here. should I start working on the check?

**Comment by romani** [comment]
Enforce is what is implied for all Checks.

`HexLiteralCaseCheck` is winner

**Comment by zyadahmed** [comment]
I am on it

**Comment by romani** [comment]
Check is merged
